openlane_run/action.yml

```diff
@@ -0,0 +1,120 @@
+name: Run design with OpenLane.
+
+description: >
+  The goal of this action is to run a given design through the OpenLane flow
+  using a given tag. Optionally a PR can be created to update a stable tag
+  file stored in the repository.
+
+inputs:
+  ol_tag:
+    description: OpenLane tag. You can set to ol_tag_file to use the value from the input.
+    default: master
+  update_tag:
+    description: If the test is successful update stable tag pointed by the ol_tag_file input.
+    default: false
+  ol_tag_file:
+    description: File where to store the new OpenLane tag
+    default: .github/openlane-stable-tag
+  tools_list:
+    description: List of the tools to update separated by a space. Example "openroad_app magic".
+    default: None
+
+runs:
+  using: composite
+  steps:
+
+    - uses: actions/checkout@v3
+      with:
+        repository: The-OpenROAD-Project/OpenLane
+        fetch-depth: 0
+
+    - uses: actions/checkout@v3
+      with:
+        path: designs/gha_design
+
+    - shell: bash
+      run: |
+        echo '::group::Export variables to use with OpenLane Makefile'
+        if [[ "${{ inputs.ol_tag }}" == "ol_tag_file" ]]; then
+          GIT_SHA="$(cat designs/gha_design/${{ inputs.ol_tag_file }})"
+        elif [[ "${{ inputs.ol_tag }}" == "master" ]]; then
+          GIT_SHA="$(git describe --exact-match --tags $(git rev-list --tags --max-count=1))"
+        else
+          GIT_SHA=${{ inputs.ol_tag }}
+        fi
+        echo "Checkout correct branch: $GIT_SHA"
+        git switch -c "$GIT_SHA" "$GIT_SHA"
+        # if we are updating any tool, we need to make sure we will use the
+        # newly generated docker image
+        if [[ "${{ inputs.tools_list }}" != "None" ]]; then
+          GIT_SHA=current
+        fi
+        echo "OPENLANE_DOCKER_TAG=${GIT_SHA}" >> $GITHUB_ENV
+        echo "OPENLANE_IMAGE_NAME=efabless/openlane:${GIT_SHA}" >> $GITHUB_ENV
+        echo "PDK_ROOT=${PWD}/pdks" >> $GITHUB_ENV
+        echo '::endgroup::'
+
+    - shell: bash
+      if: ${{ inputs.tools_list  != 'None' }}
+      run: |
+        echo '::group::Update each tool hash'
+        for tool in ${{ inputs.tools_list }}
+        do
+          python3 .github/scripts/update_tools.py "${tool}"
```

Comment:
q: reading thru https://github.com/The-OpenROAD-Project/OpenLane/blob/master/.github/scripts/update_tools.py it seems that this will get the latest version of the tools independently of what's specified in the `ol_tag`, can you confirm this is the intent?

Reply:
Yes, the goal is to use the latest version independent of `ol_tag`.